### ISSUE TITLE
zz causes flickering in the end of file with wrapped lines

### ISSUE THREAD

**Issue by AndreiNekrasOn** [karb94/neoscroll.nvim#95]
I came into this issue randomly, and was able to reproduce it only on a specific file.
When I move into the last line of the file that has wrapped lines and call `zz`, the scroll starts flickering.

![neoscroll-bug](https://github.com/karb94/neoscroll.nvim/assets/39444688/aee8dfb5-c319-4299-a17e-1b91fec1fd73)

How to reproduce:
- init.lua:
```lua
local lazypath = vim.fn.stdpath "data" .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system {
        "git",
        "clone",
        "--filter=blob:none",
        "https://github.com/folke/lazy.nvim.git",
        "--branch=stable",
        lazypath,
    }
end
vim.opt.rtp:prepend(lazypath)
require("lazy").setup {
    spec = {
        "karb94/neoscroll.nvim",
        config = function()
            require('neoscroll').setup {}
            local t = {}
            t['zz']    = {'zz', {'250'}}
            require('neoscroll.config').set_mappings(t)
        end
    },
}
```
- create file with content:
```
short line
short line
long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line 


long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line long line 

short line
short line
short line
short line
short line
short line
short line
short line
short line
short line
short line
short line
short line
short line
short line
short line
short line
```
- `G`, `zz`

I think this has something to do with `wrap` option: the bug occurs when the wrapped line ends up at the top of the screen. Unfortunately, I don't know neovim enough to debug it further.

**Comment by karb94** [comment]
I think I fixed this with 21d5297. Can you confirm?

The issue was that wrapped lines sometimes force you to scroll more lines than necessary and Neoscroll would try to backtrack which could result in an endless loop. Backtracking is not something it should do anyways so this change prevents backtracking one overshooting the target line.

**Comment by AndreiNekrasOn** [comment]
Yes, now this works as expected.
